src/cartographer/klipper/extra.py

```diff
@@ -39,43 +39,58 @@ def __init__(self, config: ConfigWrapper) -> None:
         logger.debug("Initializing Cartographer")
         self.config = KlipperCartographerConfiguration(config)
 
-        probe_config = self.config.scan_models["default"]
-        model = ScanModel(probe_config)
+        scan_config = self.config.scan_models["default"]
+        model = ScanModel(scan_config)
 
         self.mcu = KlipperCartographerMcu(config)
-        toolhead = KlipperToolhead(config)
+        toolhead = KlipperToolhead(config, self.mcu)
+
         scan_probe = ScanProbe(self.mcu, toolhead, model=model)
-        scan_endstop = ScanEndstop(self.mcu, scan_probe)
+        scan_endstop = KlipperEndstop(self.mcu, scan_probe)
+
+        touch_config = self.config.touch_models["default"]
+        touch_probe = TouchProbe(self.mcu, toolhead, touch_config)
 
-        endstop = KlipperEndstop(self.mcu, scan_endstop)
-        homing_chip = CartographerHomingChip(printer, endstop)
+        homing_chip = CartographerHomingChip(printer, scan_endstop)
 
         printer.lookup_object("pins").register_chip("probe", homing_chip)
 
+        probing_probe = touch_probe if self.config.probing_mode is ProbeMode.TOUCH else scan_probe
+
         self.gcode = printer.lookup_object("gcode")
         self._configure_macro_logger()
-        probe_macro = ProbeMacro(scan_probe)
+        probe_macro = ProbeMacro(probing_probe)
         self._register_macro(probe_macro)
-        self._register_macro(ProbeAccuracyMacro(scan_probe, toolhead))
-        query_probe_macro = QueryProbeMacro(scan_endstop, toolhead)
+        self._register_macro(ProbeAccuracyMacro(probing_probe, toolhead))
+        query_probe_macro = QueryProbeMacro(probing_probe, toolhead)
         self._register_macro(query_probe_macro)
         self._register_macro(ZOffsetApplyProbeMacro(toolhead))
 
         printer.add_object(
             "probe",
             KlipperCartographerProbe(
-                scan_probe,
-                KlipperProbeConfiguration(self.config, probe_config),
+                probing_probe,
+                self.config,
```

Comment:
`TouchProbe` has an offset of `0,0` because it is the actual nozzle measuring the distance, not the probe.